core/src/mender-api.c

```diff
@@ -65,6 +65,15 @@ static mender_err_t mender_api_http_text_callback(mender_http_client_event_t eve
  */
 static char *artifact_name = NULL;
```

Comment:
@vpodzime This variable is also shared across the two work items: check for updates and push inventory. Hence it should be protected too.